enhancements/machine-api/nutanix-multi-nics.md

```diff
@@ -0,0 +1,128 @@
+
+We will add a featue gate "NutanixMultiSubnets" (DevPreviewNoUpgrade, TechPreviewNoUpgrade) for this feature. After QE testing complete, we will add the feature gate to the "Default" feature set.
+
+```go
+// NutanixPlatformSpec holds the desired state of the Nutanix infrastructure provider.
+// This only includes fields that can be modified in the cluster.
+type NutanixPlatformSpec struct {
+  ...
+
+  // failureDomains configures failure domains information for the Nutanix platform.
+  // When set, the failure domains defined here may be used to spread Machines across
+  // prism element clusters to improve fault tolerance of the cluster.
+  // +openshift:validation:FeatureGateAwareMaxItems:featureGate=NutanixMultiSubnets,maxItems=32
+  // +listType=map
+  // +listMapKey=name
+  // +optional
+  FailureDomains []NutanixFailureDomain `json:"failureDomains"`
+}
+
+// NutanixFailureDomain configures failure domain information for the Nutanix platform.
+type NutanixFailureDomain struct {
+  ...
+
+  // subnets holds a list of identifiers (one or more) of the cluster's network subnets
+  // If the feature gate NutanixMultiSubnets is enabled, up to 32 subnets may be configured.
+  // for the Machine's VM to connect to. The subnet identifiers (uuid or name) can be
+  // obtained from the Prism Central console or using the prism_central API.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinItems=1
+  // +openshift:validation:FeatureGateAwareMaxItems:featureGate="",maxItems=1
+  // +openshift:validation:FeatureGateAwareMaxItems:featureGate=NutanixMultiSubnets,maxItems=32
+  // +openshift:validation:FeatureGateAwareXValidation:featureGate=NutanixMultiSubnets,rule="self.all(x, self.exists_one(y, x == y))",message="each subnet must be unique"
+  // +listType=atomic
+  Subnets []NutanixResourceIdentifier `json:"subnets"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+The installer should allow more than one subnets to be configured in the install-config.yaml. And pass that configuration to the installer generated Machine/MachineSet manifests when running the installer to create an OCP cluster.
+
+The Machine validation webhook should check the Nutanix providerSpec’s “subnets” field to allow more than one item and make sure there are no duplicates.
+The nutanix machine controller should allow more than one item in the NutanixMachineProviderConfig’s “subnets” field, and use this configured subnets value when creating a new VM for the Machine node.
+
+### Workflow Description
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+#### Standalone Clusters
+
+#### Single-node Deployments or MicroShift
+
+### Risks and Mitigations
+
+### Drawbacks
+
+## Test Plan
+
+- QE will test this feature
+- Will add an e2e test case for this feature
```

Comment:
What will the E2E do, and how will it differ from our existing E2E?

Reply:
added more description.